### PR TITLE
Finalize authorization, minimize requests to Schul-Cloud, fix bugs

### PR DESCRIPTION
**Pull request by MrSerth**

### CODE REVIEW

File: src/parsers/calendar/scopesToCalendarList.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 const config = require('../../config');
 
 function scopesToCalendarList(scopes, token) {
-    return {
-        links: {
-            self: `${config.ROOT_URL}/calendar/list`
-        },
-        data: scopes.map((scope) => scopeToJsonApi(scope, token))
-    };
+    return new Promise((resolve) => {
```
Comment (tamslo) [inline src/parsers/calendar/scopesToCalendarList.js]: Why do we return a promise here?

File: src/parsers/event/jsonApiToJson.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const validateJson = require('../validators/validateEventJson');
+const validJson = require('../validators/validateEventJson');
```
Comment (tamslo) [inline src/parsers/event/jsonApiToJson.js]: I personally liked the naming before better.

Comment (MrSerth) [inline src/parsers/event/jsonApiToJson.js]: I can change it back here and in many other places (this was the only one that was not aligned to the others).

Comment (tamslo) [inline src/parsers/event/jsonApiToJson.js]: Then leave it as it is, it's really not that important. Consistency ftw 😉💪

Comment (MrSerth) [inline src/parsers/event/jsonApiToJson.js]: I changed all occurrences in 04f980e.

File: src/parsers/calendar/scopesToCalendarList.js (hunk 1 above)
Comment (MrSerth) [inline src/parsers/calendar/scopesToCalendarList.js]: Removed with commit 5267806. Please compare - which do you think is better?

**Comment by MrSerth** [comment]
I incorporated your feedback, @tamslo! Thanks for your review - do you agree with my changes?

**Pull request merged**